### ISSUE TITLE
Allow unnormalized class scores for Accuracy

### ISSUE THREAD

**Issue by its-dron** [PyTorchLightning/pytorch-lightning#6140]
## 🚀 Feature
Presently, when using `Accuracy` metric on multi-class with scores (`N,C` entry in [input types](https://pytorch-lightning.readthedocs.io/en/1.2.0/extensions/metrics.html#input-types)), the scores are required to be probabilities in [0, 1]. 

However, un-thresholded accuracy can be computed without normalized probabilities as inputs, as relative ordering of scores is all that is needed. 

Given that some uses of `Accuracy` do require normalized probabilities, we could implement this as a flag that would disable the input check. 

### Motivation

It is common to work with unnormalized class scores during training, especially during classification tasks, as they are used in the more-stable `nn.CrossEntropyLoss`. Rather than having to additionally compute a softmax just for the accuracy metric, it would be reasonable to allow usage of arbitrarily scaled input data.

I specify `Accuracy` because it is the use case that I ran into, but it's possible other Metrics have the same property. 

### Pitch

Add a flag to `Accuracy` (and any other applicable metrics) that disables the input range check for `preds`.

### Alternatives

The present workaround is to apply a softmax before feeding data to your Accuracy metric.

### Additional context

https://github.com/PyTorchLightning/pytorch-lightning/blob/0456b4598f5f7eaebf626bca45d563562a15887b/pytorch_lightning/metrics/functional/accuracy.py#L25

**Comment by Jumperkables** [comment]
This please
